app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,329 @@
+    .component('notificationDrawerWrapper', {
+      templateUrl: 'views/directives/notifications/notification-drawer-wrapper.html',
+      controller: [
+        '$filter',
+        '$interval',
+        '$location',
+        '$timeout',
+        '$routeParams',
+        '$rootScope',
+        'Constants',
+        'DataService',
+        'NotificationsService',
+        'EventsService',
+        NotificationDrawerWrapper
+      ]
+    });
+
+  function NotificationDrawerWrapper(
+      $filter,
+      $interval,
+      $location,
+      $timeout,
+      $routeParams,
+      $rootScope,
+      Constants,
+      DataService,
+      NotificationsService,
+      EventsService) {
+
+      // kill switch if watching events is too expensive
+      var DISABLE_GLOBAL_EVENT_WATCH = _.get(Constants, 'DISABLE_GLOBAL_EVENT_WATCH');
+      var drawer = this;
+
+      // global event watches
+      var rootScopeWatches = [];
+      // this one is treated separately from the rootScopeWatches as
+      // it may need to be updated outside of the lifecycle of init/destroy
+      var notificationListener;
+      // our internal notifications
+      var clientGeneratedNotifications = [];
+
+      var eventsWatcher;
+      var eventsByNameData = {};
+      var eventsMap = {};
+
+      // TODO:
+      // include both Notifications & Events,
+      // rather than destroying the map each time maintain it & add new items
+
+      // final Processed set of notification groups for UI
+      // IF POSSIBLE, avoid having to convert back to an array.
+      // var notificationGroupsMap = {};
+      var notificationGroups = [];
+
+
+      var projects = {};
+
+      var getProject = function(projectName) {
+        return DataService
+                .get('projects', projectName, {}, {errorNotification: false})
+                .then(function(project) {
+                  projects[project.metadata.name] = project;
+                  return project;
+                });
+      };
+
+      var ensureProjectGroupExists = function(groups, projectName) {
+        if(projectName && !groups[projectName]) {
+          groups[projectName] = {
+            heading: $filter('displayName')(projects[projectName]) || projectName,
+            project: projects[projectName],
+            notifications: []
+          };
+        }
+      };
+
+      var deregisterEventsWatch = function() {
+        if(eventsWatcher) {
+          DataService.unwatch(eventsWatcher);
+        }
+      };
+
+      var watchEvents = function(projectName, cb) {
+        deregisterEventsWatch();
+        if(projectName) {
+          eventsWatcher = DataService.watch('events', {namespace: projectName}, _.debounce(cb, 400), { skipDigest: true });
+        }
+      };
+
+      // NotificationService notifications are minimal, they do no necessarily contain projectName info.
+      // ATM tacking this on via watching the current project.
+      var watchNotifications = function(projectName, cb) {
```

Comment:
Let's disable the notification watch until we're actually adding them to the drawer.